### ISSUE TITLE
What is the plan/roadmap for unifying SnappyData rowstore stored procedures and Spark API ? 

### ISSUE THREAD

**Issue by LarrySchumacher** [SnappyDataInc/snappydata#745]

**Comment by sumwale** [comment]
@LarrySchumacher The plan going forward is to default to Spark executions using Dataset and RDD APIs. That is both more convenient and powerful, and using from job-server REST API is simple. We have been thinking about SQL side support and will inform when something concrete is decided. If you have some ideas on this then let us know.

**Comment by LarrySchumacher** [comment]
I would like the ability to call from JDBC SQL a procedure (user defined table function?) that has access to underlying SnappySession/SnappyContext and the ability to use Spark API to return a Dataset resultset.

**Comment by sumwale** [comment]
So a way to invoke something like existing [jobs](https://snappydatainc.github.io/snappydata/programming_guide/#snappydata-jobs) from JDBC/ODBC, and get the results back as a ResultSet? Will that suffice for your use cases?

**Comment by LarrySchumacher** [comment]
yes, please!

**Comment by LarrySchumacher** [comment]
Does SnappySubmit go through the  job-server REST API or should I call  job-server REST API directly?

**Comment by jramnara** [comment]
Yes, SnappySubmit invokes the Job-server REST API. You could invoke this directly though we haven't yet made this API visible.

**Comment by jramnara** [comment]
@sumwale

**Comment by jramnara** [comment]
@LarrySchumacher You mentioned wanting a 'user define table function'... I suppose you could just use a UDF and return a Set. Doesn't this work today in Snappy ?

**Comment by jramnara** [comment]
@sumwale Instead of supporting the cumbersome SQL standard procedures (SQL IN OUT types), we should consider Higher order functions like in the Databricks platform using [TRANSFORM](https://databricks.com/blog/2017/05/24/working-with-nested-data-using-higher-order-functions-in-sql-on-databricks.html). Looks like this didn't make it into Spark.

**Comment by sumwale** [comment]
@jramnara The TRANSFORM expression is quite interesting, didn't know about it. Not clear what language the lambda functions allow for, though looks like java8.

**Comment by LarrySchumacher** [comment]
I think it would be beneficial to have the stored procedure have access to a SnappySession/SnappyContext and allowing it to return results in the form of a DataFrame/DataSet so it can participate in overall optimization of the SQL that invokes it.
